File: cakk-api/src/main/java/com/cakk/api/mapper/ShopOperationMapper.java
package com.cakk.api.mapper;

import java.util.List;

import lombok.AccessLevel;
import lombok.NoArgsConstructor;

import com.cakk.api.dto.param.operation.ShopOperationParam;
import com.cakk.domain.mysql.entity.shop.CakeShopOperation;

@NoArgsConstructor(access = AccessLevel.PRIVATE)
public class ShopOperationMapper {

	public static List<CakeShopOperation> supplyCakeShopOperationListBy(final List<ShopOperationParam> operationDays) {
		return operationDays
			.stream()
			.map(ShopOperationMapper::supplyCakeShopOperationBy)
			.toList();
	}

	public static CakeShopOperation supplyCakeShopOperationBy(final ShopOperationParam param) {
		return CakeShopOperation.builder()
			.operationDay(param.operationDay())
			.operationStartTime(param.operationStartTime())
			.operationEndTime(param.operationEndTime())
			.build();
	}
}

File: cakk-core/src/main/kotlin/com/cakk/core/mapper/ShopOperationMapper.kt
package com.cakk.core.mapper

import com.cakk.core.dto.param.shop.ShopOperationParam
import com.cakk.infrastructure.persistence.entity.shop.CakeShopOperationEntity

fun supplyCakeShopOperationListBy(operationDays: List<ShopOperationParam>): List<CakeShopOperationEntity> {
	return operationDays.map { supplyCakeShopOperationBy(it) }.toList()
}

private fun supplyCakeShopOperationBy(param: ShopOperationParam): CakeShopOperationEntity {
	return CakeShopOperationEntity.builder()
		.operationDay(param.operationDay)
		.operationStartTime(param.operationStartTime)
		.operationEndTime(param.operationEndTime)
		.build()
}
